opentelemetry-http/src/lib.rs

```diff
@@ -122,9 +134,24 @@ pub mod surf {
             }
 
             let mut response = self.send(request_builder).await?;
-            Ok(Response::builder()
+            let mut headers = HeaderMap::new();
+            for header_name in response.header_names() {
+                for header_value in &response[header_name.to_string().as_str()] {
+                    headers.append(
+                        HeaderName::from_str(&header_name.to_string())
+                            .expect("it was a header name before"),
```

Comment:
Can we convert it into `HttpError` here? Doesn't seems to right to break the application because one header cannot parse

Reply:
Done, thanks :)